egs/gigaspeech/ASR/local/compute_fbank_gigaspeech_dev_test.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# Copyright    2021  Johns Hopkins University (Piotr Żelasko)
+# Copyright    2021  Xiaomi Corp.             (Fangjun Kuang)
+#
+# See ../../../../LICENSE for clarification regarding multiple authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from pathlib import Path
+
+import torch
+from lhotse import (
+    CutSet,
+    KaldifeatFbank,
+    KaldifeatFbankConfig,
+    LilcomHdf5Writer,
+)
+
+# Torch's multithreaded behavior needs to be disabled or
+# it wastes a lot of CPU and slow things down.
+# Do this outside of main() in case it needs to take effect
+# even when we are not invoking the main (e.g. when spawning subprocesses).
+torch.set_num_threads(1)
+torch.set_num_interop_threads(1)
+
+
+def compute_fbank_gigaspeech_dev_test():
+    in_out_dir = Path("data/fbank")
+    # number of workers in dataloader
+    num_workers = 20
+
+    # number of seconds in a batch
+    batch_duration = 600
+
+    subsets = ("DEV", "TEST")
+
+    device = torch.device("cpu")
+    if torch.cuda.is_available():
+        device = torch.device("cuda", 0)
+    extractor = KaldifeatFbank(KaldifeatFbankConfig(device=device))
+
+    logging.info(f"device: {device}")
+
+    for partition in subsets:
+        cuts_path = in_out_dir / f"cuts_{partition}.jsonl.gz"
+        if cuts_path.is_file():
+            logging.info(f"{cuts_path} exists - skipping")
+            continue
+
+        raw_cuts_path = in_out_dir / f"cuts_{partition}_raw.jsonl.gz"
+
+        logging.info(f"Loading {raw_cuts_path}")
+        cut_set = CutSet.from_file(raw_cuts_path)
```

Comment:
@wgb14 you need to run `.trim_to_supervisions(keep_overlapping=False)` on this cut set in order to get data that is usable for validation (otherwise the cuts represent whole recordings). I see you have it for train cuts, but not for dev and test.

Reply:
Good catch, fixed

Reply:
Do you still have OOM after this fix?

Reply:
Yes, and I'll check with your statistic tool first.
![image](https://user-images.githubusercontent.com/49512813/146624109-eb12c8c7-b3f7-48d7-8123-0eb352bd4d65.png)
Does this mean an utterance of 22s caused the CUDA OOM?

Reply:
> Does this mean an utterance of 22s caused the CUDA OOM?

I don't think so. You need to check the total duration, after padding, in the batch that is causing CUDA OOM.

Reply:
GigaSpeech has removed all the utterances longer than 20s, the longest utterance in CutSet would be about 22.22s after speed perturbation.

Reply:
So does it still cause OOM after removing long utterances and using a smaller `--max-duration`?

Reply:
Yes, although I have set `--max-duration` to 80. By the way, does `--num-workers` affect GPU memory usage?

Reply:
> By the way, does --num-workers affect GPU memory usage?

No, I don't think so. DataLoader uses only CPU memory.

Reply:
> > Does this mean an utterance of 22s caused the CUDA OOM?
> 
> I don't think so. You need to check the total duration, after padding, in the batch that is causing CUDA OOM.

It is possible that 1) your batch size is still too large 2) it can be larger if you drop longer utterances (eg filter at 18s or less) 3) you are not using bucketing and simply found a pessimistic batch with a lot of padding (in which case you have to use bucketing or filter long cuts otherwise your training will eventually fail)